extensions/amp-story/1.0/amp-story-quiz.js

```diff
@@ -299,13 +348,130 @@ export class AmpStoryQuiz extends AMP.BaseElement {
+
+        this.hasUserSelection_ = true;
+      });
+
+      this.updateReactionData_(optionEl.optionIndex_);
+    });
+  }
+
+  /**
+   * Get the Reaction data from the datastore
+   *
+   * @return {Promise<JsonObject>}
+   * @private
+   */
+  retrieveReactionData_() {
+    return this.executeReactionRequest_()
+      .then(response => {
+        if (response.hasNoEndpoint) {
+          return;
+        }
+
+        this.handleSuccessfulDataRetrieval_(response);
+      })
+      .catch(error => {
+        dev().error(TAG, error);
+      });
+  }
+
+  /**
+   * Update the Reaction data in the datastore
+   *
+   * @param {number} reactionValue
+   * @private
+   */
+  updateReactionData_(reactionValue) {
+    this.executeReactionRequest_(reactionValue).catch(error => {
+      dev().error(TAG, error);
+    });
+  }
+
+  /**
+   * Executes a Reactions API call.
+   *
+   * @param {number} reactionValue
+   * @return {Promise<JsonObject>}
+   * @private
+   */
+  executeReactionRequest_(reactionValue) {
+    // TODO(jackbsteinberg): Add a default reactions endpoint.
+    if (!this.element.hasAttribute('endpoint')) {
+      return Promise.resolve({hasNoEndpoint: true});
+    }
+
+    let URL = this.element.getAttribute('endpoint');
+
+    const requestVars = {
+      reactionType: STORY_REACTION_TYPE_QUIZ,
+      reactionId: '', // combine url & attribute
+    };
+
+    const requestOptions = {
+      method: 'GET',
+    };
+
+    if (reactionValue !== undefined) {
+      requestVars.reactionValue = reactionValue;
+      requestOptions.method = 'POST';
+      requestOptions.body = requestVars;
+    } else {
+      URL += `?reactionType=${
+        requestVars.reactionType
+      }&reactionId=${encodeURIComponent(requestVars.reactionId)}`;
+    }
 
-      this.hasReceivedResponse_ = true;
+    return this.getClientId_().then(clientId => {
+      requestVars.clientId = clientId;
+      return this.requestService_.executeRequest(URL, null, requestOptions);
     });
   }
+
+  /**
+   * Handles incoming reaction data response
+   *
+   * RESPONSE FORMAT
+   * {
+   *  totalResponseCount: <number>
+   *  hasUserResponded: <boolean>
+   *  responses: [
+   *    {
+   *      reactionValue:
+   *      totalCount:
+   *      selectedByUser:
+   *    },
+   *    ...
+   *  ]
+   * }
+   * @param {ReactionResponseType} response
+   * @private
+   */
+  handleSuccessfulDataRetrieval_(response) {
+    this.responseData_ = {
+      totalCount: response.data.totalResponseCount,
+      data: response.data.responses,
+    };
+
+    this.hasUserSelection_ = response.data.hasUserResponded;
+    if (this.hasUserSelection_) {
+      this.quizEl_.classList.add('i-amphtml-story-quiz-post-selection');
+
+      const selectedOptionKey = this.responseData_.data.find(
+        response => response.selectedByUser
+      ).reactionValue;
+
+      this.quizEl_
+        .querySelectorAll('.i-amphtml-story-quiz-option')
+        [selectedOptionKey].classList.add(
```

Comment:
We should check to make sure `selectedOptionKey` exists in the querySelector result first.  For example, if I publish a quiz with options A, B, C, D, then you vote D, then I edit my quiz to only have options A, B, C, this will error in a way that's not meaningful (because D is no longer an option in the quiz, so you'll get something like `Cannot read property 'classList' of undefined` in the logs instead of `Quiz ID #123 does not have an option D, but the user voted option D`)